src/Repl/Decode.tsx

```diff
@@ -0,0 +1,199 @@
+  // parse by the type of the response
+  if(decoded["type"] === "Board") {
+    // Board
+    return decodeBoard(decoded["value"])
+
+  } else if(decoded["type"] === "Tuple") {
+    // Tuple
+    return decodeTuple(decoded["value"])
+
+  } else if(decoded["type"] === "Bool") {
+    // Bool
+    return decodeBool(decoded["value"])
+
+  } else if(decoded["type"] === "Int" || decoded["type"] === "Symbol") {
+    // Int || Symbol
+    return decoded["value"]
+
+  } else {
+    // fallback
+    console.error("Unable to decode value for type " + decoded["type"])
+    return "Unrecognized Value"
+
+  }
+}
+
+
+/*
+* decodeBoard
+* Takes a 2D array of JSON values, an encoded board, and
+* extracts a TSV representation of the same board to return
+*
+* board: 2D array of values that represent a BoGL board
+*/
+const decodeBoard = (board: Array<Array<JSON>>) => {
+  // map over every row in the board, and for every space decode the value
+  // join elms in rows by tabs, and join rows by line breaks
+  // double line break at front to differentiate between repeated boards in tuples
+  return "\n\n"+board.map(
+    (row) => row.map(
+      (space) => decodeValue(space[1])
+    ).join("\t")
+  ).join("\n");
+}
+
+
+/*
+* decodeTuple
+* Takes an JS array representing a tuple, which
+* can be nested repeatedly
+*
+* tuple: Array representing an encoded tuple
+*/
+const decodeTuple = (tuple: any) => {
+  // map a decode across all values, and join w/ ','s
+  return "(" + tuple.map((e) => decodeValue(e)).join(",") + ")";
+}
+
+
+/*
+ * decodeBool
+ * Decodes a Bool
+ *
+ * decoded : Decoded response that contains a bool
+ */
+const decodeBool = (decoded) => {
+  return decoded ? "True" : "False";
+}
+
+
+/*
+ * decodeExprType
+ * Decodes the type of the given expression
+ *
+ * latest : JSON obj of the latest expression
+ */
+const decodeExprType = (typ) => {
+  // report this type, if marked with ':t ' from the cmd line
+  // get first letter in lowercase
+  let fl = typ[0].toLowerCase();
+  let res = " is";
+
+  if(fl === 'a' || fl === 'e' || fl === 'i' || fl === 'o' || fl === 'u') {
+    // vowel sub y, use 'an'
+    res += " an ";
+
+  } else {
+    // consonant
+    res += " a ";
+
+  }
+  return res + typ;
+}
+
+
+/*
+ * decodeError
+ * Decodes an error related to a BoGL API request
+ *
+ * error :      Obj of error that has occurred
+ * respStatus : Status code of the response
+ */
+const decodeError = (error, respStatus : number) => {
+
+  console.error(error)
+
+  if((error instanceof SyntaxError || (error.name && error.name === "SyntaxError")) && respStatus === 504) {
+    // gateway timeout
+    return " 🤖 BoGL Says: Unable to finish running your program, or not currently online. Double check your code, or check back later! ";
+
+  } else if((error instanceof SyntaxError || (error.name && error.name === "SyntaxError"))) {
+    // bad parse error
+    return " 🤖 BoGL Says: I couldn't understand your program. Please double check it and try again! ";
```

Comment:
Note that it is possible to end up here in a normal flow of events. Try entering a string that does not parse to an expression in input mode. Coercing input strings to `Vals` can fail (see [here](https://github.com/The-Code-In-Sheep-s-Clothing/bogl/blob/597e0016483e2ad573813a045f55d14298a7b19b/src/API/JSONData.hs#L65)), which leads the back end to respond with a `SyntaxError`.

I just want to make a note of this here. I think we should discuss how to best send `input` over and possibly make changes later if needed.

Reply:
I noticed the new changes you added trip this for non-literal inputs. It should still pop off the last input when this happens, but most importantly this should never happen if we can help it. A `SyntaxError` in this category almost always means that the response from the server was not valid JSON. The actual error itself comes almost 100% from `JSON.parse(response)`. If it happens consistently, it means there's an invalid response coming from the backend, which should be rectified. When the server crashes this is the same thing that happens, and there's no way to distinguish between a legitimate failure and an actual failure if the response can't be parsed :/.